### PR TITLE
[BEAM-1220] Pass over the contribution guide

### PR DESCRIPTION
**Pull request by dhalperi**
* Mostly incubating- cleanup
* Minor prose fixes
* Fix markdown: `<code>something-<html><code>` does not properly escape `<html>`.

### CODE REVIEW

File: src/contribute/contribution-guide.md [hunk 1]
```diff
@@ -162,7 +162,7 @@ Start eclipse with a fresh workspace in a separate directory from your checkout.
 	File
 	-> Import...
 	-> Existing Maven Projects
-	-> Browse to the directory you cloned into and select incubator-beam
+	-> Browse to the directory you cloned into and select beam
```
Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: quotes around beam?

File: src/contribute/contribution-guide.md [hunk 2]
```diff
@@ -418,6 +418,6 @@ During review, committers will patch in your PR, generate the static `content/`,
 
 #### Committing website changes
 
-Follow the same committer process as above, but using `incubator-beam-site` and `asf-site`. 
+Follow the same committer process as above, but using `beam-site` and `asf-site`.
```
Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: repository and branch (ambiguous this way)

File: src/contribute/contribution-guide.md (hunk 1 above)
Comment (dhalperi) [inline src/contribute/contribution-guide.md]: done

File: src/contribute/contribution-guide.md (hunk 2 above)
Comment (dhalperi) [inline src/contribute/contribution-guide.md]: done

**Pull request closed without merging**